src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -1671,8 +1680,15 @@ MarkRanges MergeTreeDataSelectExecutor::filterMarksUsingIndex(
         {
             if (index_mark != index_range.begin || !granule || last_index_mark != index_range.begin)
                 granule = reader.read();
+            const auto * gin_filter_condition = dynamic_cast<const MergeTreeConditionGinFilter *>(&*condition);
+
+            bool result{false};
+            if (!gin_filter_condition)
+                result = condition->mayBeTrueOnGranule(granule);
+            else
+                result = gin_filter_condition->mayBeTrueOnGranuleInPart(granule, cache_in_store);
```

Comment:
Hi, is this possible that we push down the filter to inverted index and return the result rowid list directly? As inverted index is a row-level index, we have no need to scan the candidate mark ranges.